prerace_setup.py

```diff
@@ -2,52 +2,51 @@
 
 from animal_factory import factory
 from config import RACERS_CONFIG_PATH
-from constants import *
+from constants import READ, ORIGIN0_OFFSET
 
+TRACK_LENGTH = 'track_length'
+RACERS = 'racers'
+ANIMAL_TYPE = 'type'
+RACERS_CONFIG = 'racers_config'
+RACE_SETTINGS = 'race_settings'
+ROUNDS = 'rounds'
 
-def _initialize_rounds(race_settings) -> dict:
+
+def initialize_rounds(race_settings: dict) -> dict:
     rounds = {}
     for i, curr_round in enumerate(race_settings):
         racers = _build_round_racers(curr_round, race_settings)
-        _build_round(curr_round, i, race_settings, racers, rounds)
+        rounds = _build_round(curr_round, i, race_settings, racers, rounds)
     return rounds
 
 
-def _build_round(curr_round, i, race_settings, racers, rounds):
+def load_race_settings() -> dict:
+    with open(RACERS_CONFIG_PATH, READ) as cfg_file:
+        return json.load(cfg_file)[RACE_SETTINGS]
+
+
+def unpack_round_settings(curr_round: int, rounds: dict):
+    round_settings = rounds[curr_round]
+    return round_settings[RACERS], round_settings[TRACK_LENGTH]
+
+
+def _build_round(curr_round, i, race_settings: dict, racers, rounds):
     rounds[str(i + ORIGIN0_OFFSET)] = {
         TRACK_LENGTH: race_settings[curr_round][TRACK_LENGTH],
         RACERS: racers
     }
+    return rounds
```

Comment:
ok about the first two points.
About the last one, I change the racer each round as part of the design of the exercise to make the 
race as modular and configurable as possible.

Reply:
> better, but you still change a mutable object for each call. That's OK sometimes, but better to avoid it when you can.
> 
> a dict comprehension would look better. The real issue here, i think, is that the function is a bit messy. Why do you need both`race_setting` AND `curr_round`?
> 
> also, do the racer really changed every round? why do recreate them every time?

ok about the first two points.
About the last one, I change the racer each round as part of the design of the exercise to make the
race as modular and configurable as possible.